New UI: snake skin

alexnantel88 opened the pull request.

Commit:
Fix watch target for less files

Commit:
Installer - Preliminary Style Cleanup

Preliminary cleanup of installer.less. Slight modifications to symphony.forms.less to make input styles more flexible and error messages prettier.

nitriques commented on symphony/assets/css/src/installer.less:
```diff
@@ -2,42 +2,101 @@
-	margin: 0 auto;
-	padding-left: 20rem;
+	box-sizing: border-box;
+	position: relative; margin: 0 auto; padding: @spacing-smallish; max-width: @max-width-small;
```
Please use put each declaration on a new line ;)

nitriques commented on symphony/assets/css/src/symphony.forms.less:
```diff
@@ -165,10 +166,16 @@ button,
 }
 
-.submit input {
+.submit input[type="submit"] {
```
You found some input in a .submit container that was not a submit ?

Commit:
General - Normalization of all LESS files

Normalization of all Symphony core LESS files regarding comments, spacings and layout.

nitriques commented:
Cool! We might wanna keep the red background around errors ?

Commit:
Installer - Styling of links in text

Commit:
Admin - Header navigation

Add javascript events to toggle Nav states in backend.js.
Fix Nav bottom layout in admin.less.
Wrap first level text in Nav in a SPAN tag in class.administrationpage.php.

nitriques commented on symphony/assets/js/src/backend.js:
```diff
@@ -68,6 +68,96 @@
+			focusOptionEl: '#custom-toolbar .focus-option a',
+			customToolbar: '#custom-toolbar',
+			dataSourceSource: '#ds-context'
+		};
```
Please check how we define "roots" at line 57. Can you remove that big ass object an only add the one or two that you need please ?

nitriques commented on symphony/assets/js/src/backend.js:
```diff
@@ -68,6 +68,96 @@
+			return false;
+		});*/
+
+		/* Init - Show Subnav if parent active and highlight Subnav page if the case */
```
We try to never commit commented code, it's  bad practice :package:

nitriques commented on symphony/assets/js/src/backend.js:
```diff
@@ -68,6 +68,96 @@
+			$('ul > li a', t).each(function(){
+				var t = $(this);
+				if(window.location.href.indexOf(t.attr('href')) !== -1) t.parent().addClass('active');
+			});
```
This should be done in PHP. Check the `__buildNavigation()` method in class.administrationpage.php

Commit:
Admin - Change Requests Nitriques

Define JS “roots” in the proper way in backend.js.
Removal of commented code in backend.js.
Preopen-select active navigation in PHP in class.administrationpage.php.

nitriques commented on symphony/assets/js/src/backend.js:
```diff
@@ -116,38 +79,15 @@
-				$(o.navEl + '.opened').removeClass('opened');
+				$('.opened ul', Symphony.Elements.nav).slideUp(250);
+				$('.opened', Symphony.Elements.nav).removeClass('opened');
 
```
Much better don't you think ? ;)

nitriques commented on symphony/assets/js/src/backend.js:
```diff
@@ -68,6 +68,36 @@
+		// Header and Context
+		//
+		/////////////////////////////////////////////////////////////////////////// */
+
```
This can be only

```js
// Header and Context
```

Commit:
Admin - Change Requests Nitriques

Normalization of comments in backend.js.

Commit:
Admin - #btn-toggle-header-mobile

Styling and programming of the mobile button that toggles the navigation panel.

nitriques commented on symphony/lib/toolkit/class.administrationpage.php:
```diff
@@ -801,7 +805,8 @@ public function appendNavigation()
-                $xGroupLabel = new XMLElement('span', $n['name']);
+                if(is_array($n['children']) && !empty($n['children'])) $xGroupLabel = new XMLElement('span', $n['name'].Widget::SVGIcon('chevron'));
+                else $xGroupLabel = new XMLElement('span', $n['name']);
                 $xGroup->appendChild($xGroupLabel);
```
Please also use multiple lines for this. And always add curly brakets.

Commit:
Admin - Change Requests Nitriques

Append Notices to Wrapper instead of Header.
Rewrite condition with curly brackets and put array items on multiple lines.

Commit:
Admin - Update Notices code to target Wrapper

Update of the JS and LESS files to target the Notices in the Wrapper instead of in the Header.

Commit:
Admin - Creation of SVGIconContainer

Creation of the SVGIconContainer widget to wrap inputs and buttons with icons.

Commit:
Admin - Roboto Font

Download and conversion of all the necessary fonts in ordre to be able to have it offline.

Commit:
Admin - Field-Date Calendar

Rework Front-End of the calendar field.

Commit:
Admin - QA on Calendar and Tables

Small adjustments on the calendar and on the tables.
Removal of outdated JS on resize.

Commit:
Admin - Modifications on the action buttons

Remove handle if only one action possible in the page.
Change Create-New button layout to fit the main action ones.

Commit:
Admin - FileUpload target blank

Add target-blank attribute to file-upload links so they open in another tab.
Hide Notices on mobile.

Commit:
Admin - Normalize Form Inputs

Commit:
Admin - Last Tweakings before Extensions

Show fields.settings .help on mobile in symphony.forms.less.
Make links open in new tab in content.systemextensions.php.

Commit:
Admin - Beautify Selects in Safari

Commit:
Admin - Debug Affix

Commit:
Admin - Collapsible Responsive

Add resize of the collapsible instances in symphony.collapsible.js.
Fix calendar layout in filters suggestions.

Commit:
Add Session element in Header

Create Session element that will contain user links, version, and eventually languages and gravatar.

Commit:
Admin - Better Implementation of CreateButton

Add class-Featured on the CreateButton wrapper in the ContextActions.

Commit:
Style fixes to adjust to extensions

Adjust core styles to be able to take into account general extensions cases.

Commit:
Fix bug in publish data-title

Fix 2-3 things for forms

Commit:
Admin - Minor form fixes

Fix error layout.
Fix small bugs in forms.

Commit:
Frame and Form fixes for errors.

Commit:
Reposition password field help notice.

Commit:
Adjust label I tag.

Commit:
Fix corrupted font file.

Commit:
Cleanup Widget::SVGIcon

Only keep icons used in the core, the others will be added by the extensions via the registerSVGIcon method.

Commit:
Style extension statuses.

Commit:
Normalize .help and <I>

Commit:
Calculate context actions elements max-width

Commit:
Style elements with ‘label’ class like labels.

Commit:
Improve layout of select[multiple]

Commit:
Fix extension statuses.

Commit:
Skin opt groups and selectable instances

Commit:
Fix responsive affix

Make affix.js responsive.
Reorganize symphony.frames.less

Commit:
Multilingual interface fixes

Modify layout of action buttons to accept longer strings.
Removve unnecessary comments in variables.less.

Commit:
Style fixes

Fix corrupted woff2 fonts
Change font-weight from 300 to normal
Fix actions icons sizes

Commit:
Create rules for images in index tables

animaux commented:
Thanks for all the great work you guys put into this!

By permission of @nitriques I will try to give some feedback. I will also try to be as objective as possible, but as things go with design stuff there is always a great deal of things that depend more on taste than anything else.

I know this is work in progress, so please excuse me if I go too far `:)`

## Nice

+ Roboto and apparent material design inspired approach
+ Clean and minimal
+ Top-right save button and back-arrow for current section

## UI-design issues

+ Light font weight seems finicky on non-retina devices
+ Fieldset labels in caps are hard to read und get longish
+ header h1 background orb seems unneccessary
+ site titles of medium length already breaks the layout
+ input fields feel inverted, when following the brighter=more-important dogma

## Minor UI-design issues

+ Gray text contrast seems to low (already on @nitriques’ radar)
+ Fieldset legends seem oversized (when in »blue mode«)
+ Fieldset labels in caps are hard to read 

## Possibly a question of subjective taste?

+ Our customers are always very fond of the simplicity of the current UI. The fact that everything important can be found in the main horizontal nav makes it very clean. I do like the fact that the new vertical navigation shows everything at once, but the page feels much more cluttered in L-form.
+ Using lines as dividers: I feel this could lean more towards the concept of structure/layers/areas in the [material design guidelines](https://material.io/guidelines/layout/structure.html).
+ border-radius and padding seems to large on inputs and sections. Maybe we should consider [material design](https://material.io/guidelines/components/text-fields.html#text-fields-input-types) here too?

Since you have been going the Roboto-way I wonder if there aren’t a lot more design-cues from the Material design guidelines we could implement. Even though I don’t even own an android device I strongly feel that the whole visual and ux-approach seems to be a lot more consistent and thought through than f. e. iOS 7+.

Commit:
Fix responsive glitch #context

Commit:
Extensions status-error update

Commit:
Responsive fixes

Commit:
Associations - Fix layout

Commit:
Breacrumb

Hide images in breadcrumb in case first field is an ImageUpload.

Commit:
Fix black color and input spacings.

Commit:
Minor layout fixes

Commit:
Add arrows for each submenu parent

Commit:
Make whole Header scrollable

Commit:
Installer - Change Requests Nitriques

- Removal of the “quick language changer” on the second installer page in class.installerpage.php.
- Modification of the LESS layout to put each property on a separate line installer.less, symphony.forms.less, symphony.frames.less and symphony.grids.less.
- Modification of a LESS declaration targeting too specific input type in a “.submit” container in installer.less and symphony.forms.less..

Commit:
Admin - Move Datasource Source in Form

Move the DataSource Source select from the ContextBar to the actual Form.
Wrap action buttons and inputs in container elements to add SVG icons.

Commit:
Implement toggle method for vertical drawers

Commit:
Transfer TAB styles to appropriate extension

Commit:
Fix error templates

Remove unnecessary code in fatal error templates.
Modify structure to better style frame.error.

Commit:
Update error styles

- Update fatalerror.disabled with Symphony 4.x layout
- Improve error layout with 3.0.x adjustments

Commit:
Javascrip jshint validation

Commit:
Admin - Set responsive data-attributes

Set data-title attributes for all TD elements in order to reconfigure tables properly on mobile devices.

Commit:
Admin - Targets and Error Page

Style PageNotFound in symphony.frames.less.
Add Target-Blank on remaining links not pointing to Symphony.

Commit:
General big update

Update fatal error templates to put back HTML structure.
Update ExtensionManager so it verifies PHP Compatibility.
Clean LESS files.

Commit:
Rework Extensions Page

Add Dependencies column.
Put extension version with name.
Update status-error cases.

Commit:
Minor layout fixes

Fix PHP Compatibility message.
Fix file.upload i to improve image.upload layout.

Commit:
Clean installer.less

Clean unnecessary styles from installer.less.
Change message for Symphony PHP Compatibility in extension manager.

Commit:
Fix minor issues with required fields layout.

nitriques committed:
Remove __ in name function

Picked from 3e11af13d6

nitriques committed:
Simplify if Database logic

Picked from 414864a241

Commit:
Add PHP_MIN and PHP_MAX to config

Commit:
Allow 5.6 and 7.0 to fail on travis

nitriques committed:
Extract ExceptionRenderer from ExceptionHandler

We can split the responsabilities into seperate classes.

Picked from 81f3f021e4

nitriques committed:
Extract duplicated code into a function

Picked from 6eef0ca156

nitriques committed:
Move sendHeaders() into ExceptionRenderer

Picked from 78b661fb27

nitriques committed:
Make Error/Exception handler classes final

We do not need inheritance for those classes anymore

Picked from 8551614a9e

nitriques committed:
Rename specific Exception Handlers into Renderer

Classes were only implementing the render part since we do not actually
need other handlers.

Picked from 0978040cde

nitriques committed:
Fix missing rename from 72c3c9c716 in installer

GenericExceptionHandler -> ExceptionHandler
GenericErrorHandler -> ErrorHandler

Picked from d002e3b1a6

nitriques committed:
Add echos for failures

Picked from bfc7783aa5

nitriques committed:
Fix missing rename from 81f3f021e4

getTemplate() is now in ExceptionRenderer

Picked from 98b33a819e

nitriques committed:
Render shutdown errors usign their own exception

This allows us to remove duplicated code and always promote errors to
exception in order to render them

Picked from 815a5c96b2

nitriques committed:
Move all the render code into renderHtml

Picked from 7873dc9ca5

nitriques committed:
Render Shutdown exception with the proper template

Picked from 01f49b0cd6

nitriques committed:
Make all renderers implement renderHtml

Picked from dca7dc19b1

nitriques committed:
Code format

Picked from a92bb758d1

nitriques committed:
Add a text based exception output

Re #2605

This commit makes exception handler aware that it is running in the
console.

This allows us to provide text based exception display, which prevents
spewing html in the console.

Picked from 9e615c5bfb

nitriques committed:
Deprecated {$exception_type}Handler

Commit:
Touch classes removal

Removal of all .touch classes.

Commit:
Admin - Blueprints icons implemented

Implementation of SVGIcons in the whole Blueprints sections.

Commit:
Admin - Add icons in all action buttons

Commit:
Admin - Replace version information

Remove version from all pages and put it in the Header.

Commit:
Admin - Fix cantered-content layouts

Commit:
Admin - Multiple css fixes

Fix problems with form elements.
Improve the error template.

# Conflicts:
#	symphony/assets/css/src/admin.less

Commit:
Remove non-core styles from admin.less

Adjust templates to add the centered-content class when appropriate.

# Conflicts:
#	symphony/assets/css/src/admin.less

nitriques committed:
Remove all possible static and final modifiers

nitriques committed:
Fix wrong variable name

nitriques committed:
Fix private property access

nitriques committed:
Fix wrong access to now instance property $log

nitriques committed:
Fix error in Authors datasource

nitriques committed:
Remove execute() from section datasources

nitriques committed:
Make labels start with a letter (#2836)

This is needed because XML does not allow node names to start with a
letter.

Fixes #2765

Picked from 0d1ebd90a2
Picked from 88c79d776a

Rebased from b42d16f6337

nitriques committed:
Add casted version of variable()

This makes the code easier to read, because the cast info is on the
right line of code.

Commit:
Fix taglist entry_id 'key' and not 'unique'

Picked from 2b82882ec1e8b0b155367f784c862512d572a275

nitriques committed:
Add double to test

nitriques committed:
Move asProjectionList() into DatabaseQuery

This is the only place where it is used/needed.

nitriques committed:
Use $stm to create a proper sub query

nitriques committed:
Add a comment describing the regular expression

nitriques committed:
Cleanup $_REQUEST in installer also

nitriques committed:
Fix bug not calling the right render function

nitriques committed:
Fix: ExHandler needs to be check for enabled state

nitriques committed:
Replace self:$Prop by static::Method()

This allow to replace behaviour in child classes

nitriques committed:
Fix code format

nitriques committed:
Remove deprecated usage of $_COOKIE

nitriques committed:
Remove unused getcwd_safe() function

nitriques committed:
Fix code format

nitriques committed:
Remove coupling between Session and Cookie

Those two classes manages seperate things and do not need to know each
other to work properly (as long as session is started before we use the
Cookie class).

nitriques committed:
Remove $secure parameter

That value needs to be enforced for security reasons

nitriques committed:
Refactor into non-static methods

nitriques committed:
Fix code format + static call to login

nitriques committed:
Make session cookie domain strictier

This makes it possible to restrict the session cookie to the current
host only, to prevent leaking it to sub-domains.

Upgrades will stay on the old behavior.

Re #2842
Re #2843

nitriques committed:
Enable session strict domain on new install

Re #2842
Re #2843

nitriques committed:
Regenerate session id on login and logout

This should help in preventing session fixation.
This is recommended by the OWASP PHP cheat sheet.

Re #2843
Fixes #2598

nitriques committed:
Export null values as-is (#2844)

getPostdata returns null instead of 'no' on null values.
getValue returns 'None' instead of 'no' on null values.

Fixes #2692

nitriques committed:
Remove cache modifier from sub queries

nitriques committed:
Remove unsafe SQL from NavigationDatasource

nitriques committed:
Remove a couple of __ in function names

nitriques committed:
Remove the only __ in function name

nitriques committed:
Port the contribution guidelines from the wiki

From https://github.com/symphonycms/symphony-2/wiki/Contributing-to-Symphony
and https://github.com/symphonycms/symphony-2/wiki/Building-Symphony's-Assets

Fixes #2729

nitriques committed:
Add convertValuesToFilters remove fetchIDfromValue

This makes sure the feature is documented in the public API.
The name has been change to reflect better the usage.

Fixes #2171

It also fixes the broken prepropulation when the new EQFA API is used.
Finally, it checks to see if the field supports the feature before
adding anything into the query strings.

nitriques committed:
Remove repliance on class_exists to detect context

class_exists is not a valid choice anymore, because the autoloader can
play tricks on you. Also, some of them were relics of the past.

nitriques committed:
Set the name of the repo to be symphonycms

It changes all occurences of 'symphony-2'

Fixes #2373

nitriques committed:
Fix code exerpt formatting

Commit:
Fix publish-edit wiping all entry fields

nitriques committed:
Always allow generic template to be rendered

nitriques committed:
Fix broken sessions due to instance of check

Checking class_exists was not a valid option, but creating the engine
too early is not a better solution, especially since we can deduce the
engine's namespace from APP_MODE.

Commit:
Add col definition params for "char()"

Picked from f8d5b1fe

Commit:
Correct mistakes in 'where' syntax in Database doc

Picked from 24e42cb89b

nitriques committed:
Add Licence badges

nitriques committed:
Remove leading __ in private method names

Also, make them protected and call them with static::

nitriques committed:
Fix broken extensions update process

requiresUpdate() used to return the installed version, which was not
really a good API. 77d79fa9b18ff4 changed that but the update on the
code and documentation was missing.

nitriques committed:
Insert break lines and change the title

To Symphony CMS

Commit:
Add missing space before {

Commit:
Fix symphony.*.js dependancy on backend.js

Commit:
Correct Database transaction typo and add execute()->success()

nitriques committed:
Remove some line breaks

Commit:
Make Symphony errors more compact and clear.

nitriques committed:
Make getMimeType static

It is always called from a static context

nitriques committed:
Extract mergeFilePostData() into a class method

nitriques committed:
Remove duplicated key 'd.m.Y'

nitriques committed:
Make getDomain() public

Some extensions rely on it being accessible

nitriques committed:
Fix casing of XSLTProcessor and DOMDocument

nitriques committed:
Add missing call to parent::__construct()

nitriques committed:
FIx casing of the class name

nitriques committed:
Remove unused property assignment

nitriques committed:
Remove deprecated call to MySQL::cleanValue()

nitriques committed:
Fix wrong return type for filters

nitriques committed:
Move $_context into the abstract Page class

This makes sure we have the property defined in all classes

nitriques committed:
Fix casing of XSLTProcess

nitriques committed:
Fix casing of XMLElement

nitriques committed:
Fix wrong return type

nitriques committed:
Move setting the context into the Page class

This reduce code copy.
This change also makes sure that all build method share the same exact
signature.

nitriques committed:
Move $_errors property into AdministrationPage

Also make sure we put all errors in there + fix code format

nitriques committed:
Remove unused, dead assignment

nitriques committed:
Add missing property declaration

nitriques committed:
Replace is_null with empty to catch empty strings

nitriques committed:
Fix undefined $email variable

nitriques committed:
Remove undefined return value

nitriques committed:
Add abstract definition of Symphony::display()

Both engines current implement it correctly so it is safe to enforce.

nitriques committed:
Re-create deprecated $_lastResult property

This makes phpstan happy

nitriques committed:
Remove extraneous parameter

nitriques committed:
Fix wrong access to private property

Use public getter instead

nitriques committed:
Fix broken orientation() implementation

Copy paste FTW

nitriques committed:
Declare variable before use

nitriques committed:
Remove dead code

$this->_fields is not more since a long time now

nitriques committed:
Add missing parameters to parent ctor

nitriques committed:
Add $filteringForm property declaration

nitriques committed:
Remove unnecessary comment

nitriques committed:
Declare local variable $message before use

nitriques committed:
Fix undefined $section_id variable

nitriques committed:
Declare $message and $status variables before use

nitriques committed:
Check if execute is actually callable

The event API is a mess. This is only a safeguard point to help
debugging.

nitriques committed:
Move EventMessages into its own file

nitriques committed:
Add missing ENTRY_UNKNOWN constant

Referenced in SectionEvent

nitriques committed:
Remove unnecessary condition

We must always redirect, all paths are valid.

nitriques committed:
Add missing $_title property declaration

nitriques committed:
Remove dead code

nitriques committed:
Declare $errors variable before use

nitriques committed:
Enable phpstan on travis for PHP7.1

Commit:
Fix error code layout

Commit:
Fix class name in exception message

Commit:
Admin - Filter Icon

Add Filter icon to action button.
Rework inverted logic of insertAction.

Commit:
Fix content.publish, validate only if commit successful

nitriques committed:
Add logging for render exceptions

This makes it easier to debug cases where the exception handler failed
to properly handle things.

nitriques committed:
Restore recursive 404 behavior

FrontendPageNotFoundExceptionRenderer::render() is able to handle errors
while generating the 404 error page. We need to continue to foward the
calls to it.

Commit:
Fix  error message overriden with nul on field processing

Commit:
Fix association condition

Not permitting linked contents to display
Picked from be9e56c6937f1a1036628c9453790e496be249b9

Commit:
Remove unnecessary parenthesis

Commit:
Remove blue circle in #header/h1 and replace by full blue background.

Commit:
Handle entry sorting with field handle as well

Was currently usign with IDs only

Picked from 5cc2e36a48ef46d65b25ec4175789c8fe4e7a984

Commit:
Fix corrupted font files

Commit:
Put #nav arrows on the left for better reading

Commit:
Start #header and #nav rework

Commit:
Rework #header and #nav. Pre-session.

Commit:
Remove #header h1 background

Commit:
Temporary start rework session

Commit:
Temporary start rework session, almost done

Commit:
Rework Sessions in #header to make info more compact and presentable.

Commit:
Fix bug horizontal scroll in #header.

Commit:
Add toggle #session functionality.

Commit:
Session toggler hover state.

Commit:
Close session panel on blur.

Commit:
Create new page: Fix when handle already exists

Picked from c3fea5a34d2279d804ea96dfe465059bf8946498

Commit:
Fix auto-generate output-param name with handle.

Commit:
Put back dimmed state for success notice.

Commit:
Fix entries title always falling back on Untitled

Picked from f5aeea44ab8be67d9ed8b2604c46d6767c18bf76

Commit:
Fix bug of dragged page losing children

Remove force parent null if empty on page edit.

Picked from 0fa3e2e138022c37af8812bcb7cf914e2e90d273
and 9c569047017b42a02e027e2901dc483fb263c5a9

nitriques committed:
Perf: cache if fields table exists or not

This prevents us from checking for each entry.

nitriques committed:
Add AuthorLogout delegate

nitriques committed:
Fix broken limit in pagination

Limit is not relative to page. It is absolute no mather what is the
offset.

nitriques committed:
Fix broken test from last commit

nitriques committed:
Make sure we have the section_id before building

nitriques committed:
Clear $row variable on each iteration

nitriques committed:
Fix broken custom key length

nitriques committed:
Docs tidy up

- Fixed indentation + options parameter names
- Removed references to deleted classes

nitriques committed:
Add failsafe against direct calls

Commit:
Bring back section frame-header location diagram.

nitriques committed:
Revert to the old indexed by id structure

The 2.x version returned an indexed by id array.
Changing this breaks too many extensions, so we revert to the old way.

Commit:
Fix site name line-height.

Commit:
Add loading panel to avoid glitch on page load.

nitriques committed:
Add missing calls to parent constructor

nitriques committed:
Remove (forward) slashes (/) escapes in htaccess

/ shouldn't be escaped in htaccess regex expressions

Fixes #2855

Co-authored-by: michael-e <info@michael-eichelsdoerfer.de>

nitriques committed:
Update last_seen default value properly

This fix was missing from the 2.7.0 update

Re 1c074d23712ec5ed

Fixes #2594 for 3.0.0, since our requirement is to be at 2.7.0, we
cannot be sure that the 2.7.8 upgrade ran.

Commit:
Fix broken assets after rebase with Symphony 3.0.0

Commit:
Build. Fix broken assets after rebase with Symphony 3.0.0

nitriques committed:
Fix broken updater

Was broken in a4b12aed99

nitriques committed:
Make sure last_seen remains nullable upon update

Commit:
Fix transparent notices bug - default background to dark-grey.

Commit:
Build. Fix transparent notices bug - default background to dark-grey.

nitriques committed:
Fix bug where rows were being replaced

nitriques committed:
Add missing PHP_EOL after DB statement in cli mode

nitriques committed:
Add rowsGroupedByColumn()

rowsIndexedByColumn() was not the problem: the problem lied in the fact
that the function need unique values to work properly. Validation and
documentation of this fact as been added. The function will throw if the
index is already used.

To be able to work with non-unique values we can now use
rowsGroupedByColumn() which will return an array of arrays, so many
entries can live under the same group value.

Reverts 2ad8a75cc8 Fix bug where rows were being replaced

nitriques committed:
Throw if filter/sort by field name w/o a sectionid

If not, resolving the fields might lead to no data, since names are not
unique.

nitriques committed:
Make sorting entries deterministic

This commit is a port of https://github.com/symphonycms/symphony-2/pull/2872
e317a4c41cf31055dd72cbdacf7f3bccb53d407f

MySQL does not offer deterministic order bys. This leads to problems
where the user could see records move in the list or see a record that
was displayed in the previous page, which it not nice UX.

It is up to the programmer to create a deterministic sort. When the
fields contains mostly random data, the problem is not visible. But when
there are multiple records with the same value, the problem becomes
apparent: our order by clause is not unique.

The simplest fix I could find was to add the entry id as the last order
by clause. This clause needs to follow the same sort direction, to make
sure the last item becomes the first one when the order is flipped.

Fixes #2856

nitriques committed:
Make providerOf() not static

It is not called in a static context.

Re: https://github.com/symphonycms/symphony-2/pull/2874#issuecomment-425840052

nitriques committed:
Set a default log filter value when updating

Port of 93149b69cbe3fb0d280dda33fe57846a747f3147 from #2867

The update process never took care to set the default log filter value
which ignores depreciation warninngs. This commit also adds a post
update documentation to tell the user about this change.

Re #2762

nitriques committed:
Fix error introduced by code rewrite

next() always returns an entry object or null, never an array.

This problem was introduced in 154c161b39e52236054fc2bd656ecad428743de6
Re #2659

nitriques committed:
Fix broken sort by id, when id is a string

This restore sorting in the backend.

Bug introduced in 01be59c91e2d0f10b4eecbacce9e7a47a6db5472

nitriques committed:
(perf) Fix issue with tableExists()

SHOW TABLES LIKE clauses turn out to be quite slow.
It is ok to do when saving things, but should be avoided when doing
reads.

This commit falls back to the old, 2.x way of ignoring tables that do
not exists: Handle them in a catch clause. It is possible to still keep
the optimisation, where we won't repeat errors.

nitriques committed:
Ignore empty filters

They would break the SQL

nitriques committed:
Do not allow empty where clauses

Passing an empty array to a where clause created invalid SQL.
Doing so will not throw.

EntryQuery can produce empty where when a filter fails to run.
We need to make sure that we short-circuit the where creation.

nitriques committed:
Make sure we scope field fetching to the section

snyk-bot committed:
fix: package.json to reduce vulnerabilities

The following vulnerabilities are fixed with an upgrade:
- https://snyk.io/vuln/npm:extend:20180424
- https://snyk.io/vuln/npm:hoek:20180212
- https://snyk.io/vuln/npm:lodash:20180130
- https://snyk.io/vuln/npm:sshpk:20180409
- https://snyk.io/vuln/npm:stringstream:20180511
- https://snyk.io/vuln/npm:underscore.string:20170908

Commit:
First import of new UI (for 4.x)

This is the import made from the exploration made in 5a61bf5...46583b4
by Alexandra Nantel (@alexnantel88) and Marc-André Charpentier (@BloodBrother)

It introduces less as the source code language for the css layer.

The gruntfile has been updated accordingly.
